chains/evm/calls/events/listener.go

```diff
@@ -30,8 +35,14 @@ func NewListener(client ChainClient) *Listener {
 }
 
 func (l *Listener) FetchDeposits(ctx context.Context, contractAddress common.Address, startBlock *big.Int, endBlock *big.Int) ([]*Deposit, error) {
```

Comment:
I think the root span should be in the deposit event handler and the context should be passed in the FetchDeposits and HandleDeposit functions so they are child spans.
Please check everything that is missing a context so we can have proper parent-child spans.

Reply:
oh it is actually there but in Sygma Repo, I guess the main confusion here is that code is spread among two repos actually) 
Although I will add the same logic here as well